### PR TITLE
Replace Rc<RefCell<Response>> with Rc<Response>

### PR DESCRIPTION
**Pull request by nikkisquared**
Similar to why I'm doing #9172, I'm changing `Rc<RefCell<Response>>` into `Rc<Response>` and wrapping fields in `RefCell<T>` as needed. So far I've only needed to change `url_list`, but I'm sure I'll need to update more fields in the future.

There's two lines that don't compile that I'm not going to bother trying to fix until #9172 is approved, because getting them to work now will just require more changes after merging with the changes #9172 makes to them.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9195)

### CODE REVIEW

**Comment by nikkisquared** [comment]
This is no longer dependent on #9172. How's it looking for merging?

**Comment by jdm** [comment]
It will still need to be rebased after #9172 unfortunately.

**Comment by nikkisquared** [comment]
@jdm Right, that was mis-worded on my part. I just meant how there's no longer any temporarily commented out lines required for the PR. How do changes themselves look?

**Comment by jdm** [comment]
Ah. In that case they look fine to me.

**Comment by nikkisquared** [comment]
Great! I'll just hope that the bot gets to the other PR soon so this can get put on the stack :)

**Comment by nikkisquared** [comment]
I've rebased against master and fixed the merge conflicts, so it should be good to go now!

File: components/net/fetch/request.rs [hunk 1]
```diff
@@ -531,7 +529,7 @@ fn http_fetch(request: Rc<Request>,
         _ => { }
     }
 
-    let response = response.into_inner();
+    // let response = response;
```
Comment (jdm) [inline components/net/fetch/request.rs]: Let's remove this line.

**Comment by jdm** [comment]
With that change, this is good to merge.

**Comment by nikkisquared** [comment]
@jdm done!

**Pull request merged**